frontend/pages/schedule/ManageSchedulePage/components/ScheduleEditorModal/ScheduleEditorModal.tsx

```diff
@@ -35,37 +37,56 @@ interface IFormData {
 interface IScheduleEditorModalProps {
   allQueries: IQuery[];
   onCancel: () => void;
-  onScheduleSubmit: (formData: IFormData) => void;
+  onScheduleSubmit: (
+    formData: IFormData,
+    editQuery: IGlobalScheduledQuery | undefined
+  ) => void;
+  editQuery?: IGlobalScheduledQuery;
 }
 interface INoQueryOption {
   id: number;
   name: string;
 }
 
+const generateLoggingType = (query: IGlobalScheduledQuery) => {
+  if (query.snapshot) {
+    return "snapshot";
+  }
+  if (query.removed) {
+    return "differential";
+  }
+  return "differential_ignore_removals";
+};
+
 const ScheduleEditorModal = ({
   onCancel,
   onScheduleSubmit,
   allQueries,
+  editQuery,
 }: IScheduleEditorModalProps): JSX.Element => {
   const [showAdvancedOptions, setShowAdvancedOptions] = useState<boolean>(
     false
   );
   const [selectedQuery, setSelectedQuery] = useState<
     IGlobalScheduledQuery | INoQueryOption
   >();
-  const [selectedFrequency, setSelectedFrequency] = useState<number>(86400);
+  const [selectedFrequency, setSelectedFrequency] = useState<number>(
+    editQuery ? editQuery.interval : 86400
+  );
   const [
     selectedPlatformOptions,
     setSelectedPlatformOptions,
-  ] = useState<string>("");
+  ] = useState<string>(editQuery?.platform ? editQuery?.platform : "");
```

Comment:
A more efficient way: `editQuery?.platform || ""`

Reply:
OMG thank you!